pkg/cloudprovider/providers/azure/azure_wrap.go

```diff
@@ -54,25 +56,74 @@ func ignoreStatusNotFoundFromError(err error) error {
 	return err
 }
 
+// cache used by getVirtualMachine
+// 15s for expiration duration, 10m for proactive cleanup interval
+var vmCache = cache.New(15*time.Second, 10*time.Minute)
```

Comment:
Why use this over [client-go's expiration cache](https://github.com/kubernetes/client-go/blob/master/tools/cache/expiration_cache.go)?

Reply:
yeah, seems better to use a library we're already using...  thanks!

Reply:
Thanks for pointing out. Updated.